### PR TITLE
[MM-20462] migrate any_team_permission_gate to typescript

### PR DESCRIPTION
**Pull request by Tak-Iwamoto**
#### Summary
- Migrate  'components/permissions_gates/any_team_permission_gate' module and associated tests to TypeScript
#### Ticket Link
Fixes https://github.com/mattermost/mattermost-server/issues/15478
JIRA: https://mattermost.atlassian.net/browse/MM-20462

### CODE REVIEW

File: components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+        invert: false,
+    }
+
+    render() {
+        if (this.props.hasPermission && !this.props.invert) {
+            return this.props.children;
+        }
+        if (!this.props.hasPermission && this.props.invert) {
+            return this.props.children;
+        }
```
Comment (Willyfrog) [inline components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx:40]: this is basically a XOR operation, we can simplify it like
```suggestion
        if (this.props.hasPermission !== this.props.invert) {
            return this.props.children;
        }
```

**Review by Willyfrog — APPROVED** [review]
looking good, left a small improvement, but it is not a blocker for this

File: components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx (hunk 1 above)
Comment (Tak-Iwamoto) [reply components/permissions_gates/any_team_permission_gate/any_team_permission_gate.tsx]: Thanks ! I committed your suggestion.

**Review by ashishbhate — APPROVED** [review]

**Review by jgilliam17 — APPROVED** [review]
Thank you @Tak-Iwamoto 
Tested, looks good to merge.
- Verified team options based on user's permissions - as expected.

**Pull request merged**